File: src/main/java/com/brandlive/qa/automation/events3/pages/admin/registrationsetting/ProfileInfo.java
package com.brandlive.qa.automation.events3.pages.admin.registrationsetting;

import com.brandlive.qa.automation.utils.Base;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

public class ProfileInfo extends Base {

    public void addProfileInfoPage() {
        String profileToggleOff = "(//div[@class='toggle-card']//div[@class='form-control switch'])[1]";
        WebElement profileToggleOffButton = waitForDisplayedAndEnabled(By.xpath(profileToggleOff), "Profile Toggle on the registration page.");
        if (profileToggleOffButton.isDisplayed()) {
            profileToggleOffButton.click();
            log.info("Enabled the Profile toggle");
        }
        sleep(3);
    }
}

File: test/page-objects/streams/UI/profileInfo.ts
import Page from "../../page"
import reporter from "../../../utils/reporter"

class ProfileInfo extends Page {

    constructor() {
        super()
    }

    // Registration profile info toggle details
    get profileInfoToggleON() { return $(`(//label[text()='Profile Info']//following::div[@class='form-control switch on'])[1]`) }
    get profileInfoToggleOFF() { return $(`(//label[text()='Profile Info']//following::div[@class='form-control switch'])[1]`) }

    
    async profileInfoToggleAction(toggle: string) {
        try {
            if (toggle.match("ON")) {
                await this.click(await this.profileInfoToggleOFF)
                reporter.addStep("info", 'Enabled the Profile Info toggle.')
            } else {
                await this.click(await this.profileInfoToggleON)
                reporter.addStep("info", 'Disabled the Profile Info toggle.')
            }
        } catch (err) {
            err.message = `Error performing ${toggle} action on the Profile Info toggle: , ${err.message} `
            throw err
        }
    }
}
export default new ProfileInfo